src/tests/system-env/ARM/t_system_env_ARM_hwmon.cpp

```diff
@@ -0,0 +1,187 @@
+    values.push_back(2); /* big */
+    values.push_back(3); /* LITTLE */
+    values.push_back(4); /* GPU */
+
+    return values;
+}
+
+int is_file_exist(const char *file)
+{
+    struct stat statbuf;
+    return stat(file, &statbuf);
+}
+
+int check_user_read_permissions(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    if (statbuf.st_mode & S_IRUSR)
+    {
+        return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+int check_user_write_permissions(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    if (statbuf.st_mode & S_IWUSR)
+    {
+        return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+int check_group_read_permissions(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    if (statbuf.st_mode & S_IRGRP)
+    {
+        return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+int check_group_write_permissions(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    if (statbuf.st_mode & S_IWGRP)
+    {
+        return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+int check_user_id(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    return statbuf.st_uid;
+}
+
+int check_group_id(const char *file)
+{
+    struct stat statbuf;
+    if (stat(file, &statbuf))
+    {
+        return EXIT_FAILURE;
+    }
+    return statbuf.st_gid;
+}
+
+class HwmonTest : public ::testing::TestWithParam<int>
+{
+};
+
+TEST_P(HwmonTest, Exists)
+{
+    int testparam = GetParam();
+    char *filename;
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/power%d_input", testparam);
+    EXPECT_EQ(0, is_file_exist(filename));
+    free(filename);
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/temp%d_input", testparam);
+    EXPECT_EQ(0, is_file_exist(filename));
+    free(filename);
+}
+
+TEST_P(HwmonTest, UserID)
+{
+    int testparam = GetParam();
+    char *filename;
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/power%d_input", testparam);
+    /* Check if user owner is root */
+    EXPECT_EQ(0, check_user_id(filename));
+    free(filename);
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/temp%d_input", testparam);
+    /* Check if user owner is root */
+    EXPECT_EQ(0, check_user_id(filename));
+    free(filename);
+}
+
+TEST_P(HwmonTest, UserPerms)
+{
+    int testparam = GetParam();
+    char *filename;
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/power%d_input", testparam);
+    EXPECT_EQ(1, check_user_permissions(filename));
+    free(filename);
+
+    asprintf(&filename, "/sys/class/hwmon/hwmon0/temp%d_input", testparam);
+    EXPECT_EQ(1, check_user_permissions(filename));
+    free(filename);
+}
+
+TEST_P(HwmonTest, GroupID)
```

Comment:
Is having group permissions to hwmon required? Or is just user level permissions sufficient? We have the group checks for msr-safe, since we deploy with the msr group.

Reply:
Delete the two group tests.